apiary.apib

```diff
@@ -0,0 +1,397 @@
+FORMAT: 1A
+HOST: http://deploy.cwp.govt.nz/naut
+
+# Deploynaut API
+
+This API is a newly developed interface for the CWP Dashboard. Data structure is based on the
+[JSON API specifications](http://jsonapi.org/).
+
+
+# Using the API [/]
+
+In order to use the API, you will need to acquire a token via Service Desk request. Any request you make to
+an endpoint should include the following:
+
+## Making a request: Headers [GET]
+
+These headers will authenticate you against Deploynaut to evaluate whether you have the permissions necessary
+to access the information you are requesting. You can send a GET request to www.deploy.cwp.govt.nz/naut to
+test that you are authenticating correctly - it will return your user information.
+
++ Request (application/json)
+
+    + Headers
+
+            X-api-version: 1.0
+            X-user-email: sherbert.lemon@hogwarts.uk
+            X-user-token: fa88787fcd99787
+            Accept: application/json
+            Host: deploy.cwp.govt.nz
+
+
++ Response 200 (application/json)
+
+        {
+            "data": [
+                {
+                    "type": "users",
+                    "id": "2379",
+                    "name": "Albus Dumbledore",
+                    "email": "sherbert.lemon@hogwarts.uk",
+                    "phone": "0203040506"
+                }
+            ]
+        }
+
+
+# Stacks [/projects]
+
+## List all stacks [GET /projects]
+
+Will return a list of all stacks that you have the authority to view, with relevant datapoints.
+
++ Response 200 (application/json)
+
+        {
+            "count": 1,
+            "data": [
+                {
+                    "type": "stacks",
+                    "id": "CWP001",
+                    "name": "CWP Stack 001",
```

Comment:
Department of Internal Affairs (agency) has an instance called Public Sector Intranet (name) which is known as betapsi (id). In CWP parlance.

Title and Name are such interchangeable terms, this seems bad. ID is also a bit odd. That will probably have to be figured out in the Great Renaming of 2017. DNProject only seems to have 'Name'. Hrmph.

I agree that name seems better than id here. Will re-jig.